### DISCUSSION TITLE
Migrate from bit.cloud to locally developed components

### DISCUSSION THREAD

**Discussion by anujkumar-df** [teambit/bit#9230] [Help]
What changes do I need to make so that we don't rely on bit.cloud? I suppose I need to make changes in .bitmap file and workspaces.jsonc, but does anyone have an idea what to do?

**Comment by anujkumar-df** [comment]
An example of how my .bitmap file looks like:

```
    "base-app/renderers/label": {
        "scope": "tiui.remote",
        "version": "0.0.50",
        "mainFile": "index.ts",
        "rootDir": "remote/base-app/renderers/label"
    },
    "base-app/renderers/simple-dialog": {
        "scope": "tiui.remote",
        "version": "0.0.77",
        "mainFile": "index.ts",
        "rootDir": "remote/base-app/renderers/simple-dialog"
    },
    "configs/tiui-formatter": {
        "scope": "tiui.remote",
        "version": "0.0.3",
        "mainFile": "index.ts",
        "rootDir": "remote/configs/tiui-formatter"
    },
```

**Comment by itaymendel** [comment]
hi, if you are looking to self-host your components, follow this guide - https://github.com/teambit/bit/discussions/4707
and also from the docs - https://bit.dev/reference/reference/scope/running-a-scope-server

bit is fully open source, and you can host scopes on your own infrastructure. this will require setting up dedicated containers, storage and a registry. also, most bit-cloud features are not open source.
